zuul-core/src/main/java/com/netflix/zuul/message/ZuulMessageImpl.java

```diff
@@ -167,7 +168,7 @@ public String getBodyAsText() {
     @Override
     public int getBodyLength() {
         int size = 0;
-        for (final HttpContent chunk : bodyChunks) {
+        for (final HttpContent chunk : ImmutableList.copyOf(bodyChunks)) {
```

Comment:
Immutable list is just going to iterate through it as well, I dont think this would fix it.